templates/s3-bucket-v2.j2

```diff
@@ -175,6 +178,16 @@ Resources:
               - "s3:ListBucket*"
               - "s3:GetBucketLocation"
             Resource: !If [EnableEncryption, !GetAtt SynapseEncryptedExternalBucket.Arn, !GetAtt SynapseExternalBucket.Arn]
+          -
+            Sid: "RequireCanonicalIdOnObjectUpdates"
+            Effect: Deny
+            Principal:
+              AWS: !Ref GrantAccess
```

Comment:
@zaro0508  I think this logic should be changed.  We should _not_ deny `PutObject` access to _all_ in the 'principals' list if they don't add the `bucket-owner-full-control` ACL.  Instead we should only apply this condition to principals who are **not** in the account that owns the bucket.  I.e., if a principal in the account that owns the bucket uploads a file without the ACL that is OK because the bucket owner *already* owns the uploaded object.    This change is important because, as @BrunoGrandePhD discovered for us, when uploading through the S3 console there is no way to add an ACL to the created object.

Reply:
I think i have done what is requested.  I used an example from aws docs to come up with the configs,  https://aws.amazon.com/premiumsupport/knowledge-center/explicit-deny-principal-elements-s3/

Reply:
It looks like the change you made addresses the second issue described above (omitting Synapse from the constraint) but does not address the first issue (omitting principals in the account that owns the bucket).  Do you agree?

Reply:
Done.  The AWS account owns the bucket.  Cloudformation does not have a way to reference account IDs therefore the user will need to pass the ID in.  Info on how to get the user ID for an AWS account is here https://docs.aws.amazon.com/AmazonS3/latest/userguide/finding-canonical-user-id.html  it will look something like this..
```
➜   aws s3api list-buckets --query Owner.ID --output text
9038e06f22b4c2611873a9ac491ce754aa2353b45e19ab508577ee99863128ed
```